lib/Delighted/ListResource.php

```diff
@@ -0,0 +1,63 @@
+<?php
+
+namespace Delighted;
+
+class ListResource
+{
+    private $klass;
+    private $path;
+    private $opts;
+    private $client;
+    private $iteration_count;
+    private $done;
+    private $next_link;
+
+    public function __construct($klass, $path, $opts, $client)
+    {
+        $this->klass = $klass;
+        $this->path = $path;
+        $this->opts = $opts;
+        $this->client = $client;
+        $this->iteration_count = 0;
+        $this->done = false;
+    }
+
+    public function autoPagingIterator($opts = []) {
+        if ($this->done) {
+            throw new PaginationException("pagination completed");
+        }
+
+        $auto_handle_rate_limits = $opts['auto_handle_rate_limits'] ?? false;
+
+        while (true) {
+            try {
+                // Get first (or next) page
+                if ($this->iteration_count == 0) {
+                    $response = $this->client->get_request($this->path, $this->opts);
+                } else {
+                    $response = $this->client->get_request($this->next_link);
+                }
+            } catch (\Delighted\RequestException $e) {
+                if ($auto_handle_rate_limits && !empty($e->getRetryAfter())) {
+                    // Sleep and retry call
+                    sleep($e->getRetryAfter());
+                    continue;
```

Comment:
Does the `continue` retry the `try` block?

Reply:
Keyword itself does not, but pretty similar to the Python version, the parent `while` block only encompasses that `try` block, so continuing from there, will effectively retry the `try` block.

Reply:
Got it. I think I originally misread the whole block going all the way to the bottom, but this makes sense. Thanks for clarifying that!